src/main/scala/zio/prelude/Associative.scala

```diff
@@ -76,102 +158,228 @@ object Associative extends Lawful[AssociativeEqual] {
+   * values.
+   */
+  implicit val DoubleProdCommutativeIdentity: Commutative[Prod[Double]] with Identity[Prod[Double]] =
+    new Commutative[Prod[Double]] with Identity[Prod[Double]] {
+      def combine(l: => Prod[Double], r: => Prod[Double]): Prod[Double] = Prod(l * r)
+      val identity: Prod[Double]                                        = Prod(1)
+    }
 
-  implicit val DoubleSumAssociative: Associative[Sum[Double]] =
-    make[Sum[Double]]((l, r) => Sum(l + r))
+  /**
+   * The `Commutative` and `Identity` instance for the product of `Double`
+   * values.
+   */
+  implicit val DoubleSumCommutativeIdentity: Commutative[Sum[Double]] with Identity[Sum[Double]] =
+    new Commutative[Sum[Double]] with Identity[Sum[Double]] {
+      def combine(l: => Sum[Double], r: => Sum[Double]): Sum[Double] = Sum(l + r)
+      val identity: Sum[Double]                                      = Sum(0)
+    }
 
+  /**
+   * The `Associative` instance for the first of `A` values.
+   */
   implicit def firstAssociative[A]: Associative[First[A]] =
     make((l: First[A], _: First[A]) => l)
 
-  implicit val FloatProdAssociative: Associative[Prod[Float]] =
-    make[Prod[Float]]((l, r) => Prod(l * r))
+  /**
+   * The `Commutative` and `Identity` instance for the product of `Float`
+   * values.
+   */
+  implicit val FloatProdCommutativeIdentity: Commutative[Prod[Float]] with Identity[Prod[Float]] =
+    new Commutative[Prod[Float]] with Identity[Prod[Float]] {
+      def combine(l: => Prod[Float], r: => Prod[Float]): Prod[Float] = Prod(l * r)
+      val identity: Prod[Float]                                      = Prod(1)
+    }
 
-  implicit val FloatSumAssociative: Associative[Sum[Float]] =
-    make[Sum[Float]]((l, r) => Sum(l + r))
+  /**
+   * The `Commutative` and `Identity` instance for the sum of `Float`
+   * values.
+   */
+  implicit val FloatSumCommutativeIdentity: Commutative[Sum[Float]] with Identity[Sum[Float]] =
+    new Commutative[Sum[Float]] with Identity[Sum[Float]] {
+      def combine(l: => Sum[Float], r: => Sum[Float]): Sum[Float] = Sum(l + r)
+      val identity: Sum[Float]                                    = Sum(0)
+    }
 
-  implicit val IntProdAssociative: Associative[Prod[Int]] =
-    make[Prod[Int]]((l, r) => Prod(l * r))
+  /**
+   * The `Commutative` and `Identity` instance for the product of `Int` values.
+   */
+  implicit val IntProdCommutativeIdentity: Commutative[Prod[Int]] with Identity[Prod[Int]] =
+    new Commutative[Prod[Int]] with Identity[Prod[Int]] {
+      def combine(l: => Prod[Int], r: => Prod[Int]): Prod[Int] = Prod(l * r)
+      val identity: Prod[Int]                                  = Prod(1)
+    }
 
-  implicit val IntSumAssociative: Associative[Sum[Int]] =
-    make[Sum[Int]]((l, r) => Sum(l + r))
+  /**
+   * The `Commutative` and `Identity` instance for the sum of `Int` values.
+   */
+  implicit val IntSumCommutativeIdentity: Commutative[Sum[Int]] with Identity[Sum[Int]] =
```

Comment:
shouldn't this be `Inverse[Sum[Int]]`? (analogously for other `Sum`s of numbers)

Reply:
Great point! Will change.